src/interface/term.ml

```diff
@@ -1007,3 +1007,155 @@ module type Smtlib_Bitv = sig
   val bvsge : t -> t -> t
   (** Boolean signed arithmetic comparison (greater or equal than). *)
 end
+
+module type Smtlib_Float = sig
+
+  (** Floating points are complicated so this documentation is not in anyway
+      sufficient. A detailed description of the theory together with the
+      rationale of several models decisions as well as a formal semantics of the
+      theory can be found in
+
+      [BTRW15] Martin Brain, Cesare Tinelli, Philipp Ruemmer, and Thomas Wahl. An
+      Automatable Formal Semantics for IEEE-754 Floating-Point Arithmetic
+      Technical Report, 2015. (http://smt-lib.org/papers/BTRW15.pdf)
+  *)
+
+  type t
+  (** The type of terms *)
+
+  val fp : string -> string -> string -> t
+  (** Construct a floating point from bitvector literals (sign, exponent, significand) *)
+
+  val roundNearestTiesToEven: t
+  (** constant for rounding mode RNE *)
+
+  val roundNearestTiesToAway: t
+  (** constant for rounding mode RNA *)
+
+  val roundTowardPositive: t
+  (** constant for rounding mode RTP *)
+
+  val roundTowardNegative: t
+  (** constant for rounding mode RTN *)
+
+  val roundTowardZero: t
+  (** constant for rounding mode RTZ *)
+
+  val plus_infinity: int -> int -> t
+  (** The constant plus infinity, it is also equivalent to a literal *)
```

Comment:
These names are also problematic to have at the root of the `Expr` module, as they could realistically want to be used by other arithmetic theories. Given the high number of such occurrences in the float signature, I think a dedicated sub-module in `Expr` would be the best solution (given that renaming most or all the problematic example with `fp_` is pretty much the same as having things prefixed with `Float.`).
(note: sorry for some comments which happened a bit not chronologically with how they'll appear on github, ^^)